nomic/data_operations.py

```diff
@@ -846,3 +846,90 @@ def remove(self, ids: List[str], tags: List[str], delete_all: bool = False) -> b
 
     def __repr__(self) -> str:
         return str(self.df)
+
+
+class AtlasMapData:
+    """
+    Atlas Map Metadata State
+    """
+
+    def __init__(self, projection: "AtlasProjection"):
+        self.projection = projection
+        self.project = projection.project
+        self.id_field = self.projection.project.id_field
+        # Run fetch_tiles first to guarantee existence of quad feather files
+        try:
+            self._tb: pa.Table = self.projection._fetch_tiles()
+            sidecars, _ = self._download_data()
+            self._fetch_tiles_with_all_sidecars(sidecars)
+
+        except pa.lib.ArrowInvalid as e:
+            raise ValueError("Failed to fetch tiles for this map")
+        
+    def _fetch_tiles_with_all_sidecars(self, additional_sidecars = None):
+        tbs = []
+        root = feather.read_table(self.projection.tile_destination / "0/0/0.feather")
+        try:
+            small_sidecars = set([v for k, v in json.loads(root.schema.metadata[b'sidecars']).items()])
+        except KeyError:
+            small_sidecars = []
+        for path in self.projection._tiles_in_order():
+            tb = pa.feather.read_table(path).select(["_id"])
+            for sidecar_file in small_sidecars:
+                carfile = pa.feather.read_table(path.parent / f"{path.stem}.{sidecar_file}.feather")
+                for col in carfile.column_names:
+                    tb = tb.append_column(col, carfile[col])
+            for big_sidecar in additional_sidecars:
+                fname = base64.urlsafe_b64encode(big_sidecar.encode('utf-8')).decode("utf-8")
+                carfile = pa.feather.read_table(path.parent / f"{path.stem}.{fname}.feather")
```

Comment:
Probably want to add `memory_map = True` here. These files will get *big*.

Reply:
Added. 
Question: all the data contents for each feather file are sent over an http request and read into a pyarrow table as well for writing to disk. Are there concerns there as well?

Reply:
They should be one at a time, which hopefully won't be an issue. The issue is if we read them all at once.